File: core/commands/registry/irules/xml__disable.py
# Enriched from F5 iRules reference documentation.
"""XML::disable -- Changes the XML plugin from full patching mode to passthrough."""

from __future__ import annotations

from ....compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget
from .._base import CommandDef
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from ..namespace_models import EventRequires
from ..signatures import Arity
from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/XML__disable.html"


@register
class XmlDisableCommand(CommandDef):
    name = "XML::disable"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="XML::disable",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="Changes the XML plugin from full patching mode to passthrough.",
                synopsis=("XML::disable",),
                snippet="Changes the XML plugin from full patching mode to passthrough.",
                source=_SOURCE,
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="XML::disable",
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.STREAM_PROFILE,
                    writes=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )

File: rust/tcl-registry/src/commands/irules/xml__disable.rs
//! `XML::disable` iRules command.
use crate::prelude::*;
pub fn spec() -> CommandSpec {
    CommandSpec {
        name: "XML::disable",
        dialects: Some(DialectSet::IRULES),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet {
            summary: "Changes the XML plugin from full patching mode to passthrough.",
            synopsis: &["XML::disable"],
            snippet: "Changes the XML plugin from full patching mode to passthrough.",
            source: "https://clouddocs.f5.com/api/irules/XML__disable.html",
            examples: "",
            return_value: "",
        }),
        forms: &[FormSpec {
            kind: FormKind::Default,
            synopsis: "XML::disable",
        }],
        ..CommandSpec::DEFAULT
    }
}
